heron/packing/tests/java/com/twitter/heron/packing/PackingUtilsTest.java

```diff
@@ -37,6 +54,36 @@
     return containerPlan;
   }
 
+  private static PackingPlan generatePacking(Map<Integer, List<InstanceId>> basePacking) {
+    Resource resource = new Resource(2.0, 6 * Constants.GB, 25 * Constants.GB);
+
+    Set<PackingPlan.ContainerPlan> containerPlans = new HashSet<>();
+
+    for (int containerId : basePacking.keySet()) {
+      List<InstanceId> instanceList = basePacking.get(containerId);
+
+      Set<PackingPlan.InstancePlan> instancePlans = new HashSet<>();
+
+      for (InstanceId instanceId : instanceList) {
+        String componentName = instanceId.getComponentName();
+        Resource instanceResource;
+        if ("bolt".equals(componentName)) {
```

Comment:
can we `switch` on componentName and have an explicit case for "bolt" and "spout" and fail on default?